patches/net/minecraft/world/entity/npc/Villager.java.patch

```diff
@@ -37,12 +37,14 @@
                  p_35409_.addFreshEntityWithPassengers(witch);
                  this.releaseAllPois();
                  this.discard();
-@@ -831,7 +_,7 @@
+@@ -831,7 +_,9 @@
      @Override
      public boolean wantsToPickUp(ItemStack p_35543_) {
          Item item = p_35543_.getItem();
 -        return (WANTED_ITEMS.contains(item) || this.getVillagerData().getProfession().requestedItems().contains(item))
-+        return (WANTED_ITEMS.contains(item) || this.getVillagerData().getProfession().requestedItems().contains(item) || (p_35543_.getItem() instanceof net.neoforged.neoforge.common.SpecialPlantable specialPlantable && specialPlantable.villagerCanPlantItem(this)))
++        boolean isFarmerDesiredSeed = (p_35543_.is(ItemTags.VILLAGER_PLANTABLE_SEEDS) || (p_35543_.getItem() instanceof net.neoforged.neoforge.common.SpecialPlantable specialPlantable && specialPlantable.villagerCanPlantItem(this)))
++                && this.getVillagerData().getProfession().secondaryPoi().stream().anyMatch(secondaryPoi -> secondaryPoi.defaultBlockState().is(net.neoforged.neoforge.common.Tags.Blocks.VILLAGER_FARMLANDS));
++        return (WANTED_ITEMS.contains(item) || this.getVillagerData().getProfession().requestedItems().contains(item) || isFarmerDesiredSeed) // Neo: Fixes MC-274244
```

Comment:
Update the comment here in the following ways
1. Move it to the top of the hunk instead of inline on the last return
2. Note the changes required to support `SpecialPlantable`, and why they are being made here
3. Note that `MC-274244` is fixed, why it is being fixed, and what part of the patch actually fixes the issue.

Reply:
Done now in https://github.com/neoforged/NeoForge/pull/1256/commits/cfbfba3071be818a103041c243ea5d3e364b3b4c